build.sbt

```diff
@@ -88,7 +87,10 @@ lazy val commonSettings = Seq(
   scalacOptions ++= lintFlags.value,
 
   scalacOptions in Test ~= { _.filterNot(_.contains("-Ywarn-unused")) },
-  scalacOptions in Test += "-Xmacro-settings:materialize-derivations",
+  scalacOptions in Test ++= forScalaVersions {
+    case (2, _) => List("-Xmacro-settings:materialize-derivations")
```

Comment:
Does Scala 3 complain if you use this flag? If not I'd leave it set on all version for simplicity (`-Xmacro-settings:` are reserved for use by macros anyway).

Reply:
We get a warning:

```
[warn] bad option '-Xmacro-settings:materialize-derivations' was ignored
```

This is eventually going away with the deprecation of the `Derivation` type, so I wouldn't worry too much about it.